client/web/wasm/src/viewport.rs

```diff
@@ -16,22 +16,35 @@ pub fn select_tool(tool: String) -> Result<(), JsValue> {
 	})
 }
 
-static mut POS: (u32, u32) = (0, 0);
-
 /// Mouse movement with the bounds of the canvas
 #[wasm_bindgen]
 pub fn on_mouse_move(x: u32, y: u32) {
-	// SAFETY: This is safe because the code can only ever run in a single thread
-	unsafe {
-		POS = (x, y);
-	}
+	EDITOR_STATE.with(|editor| {
+		let mut editor = editor.borrow_mut();
+		if editor.tools.mouse_is_clicked {
+			editor.tools.trace.append_point(x, y)
+		}
```

Comment:
I think it might be useful to save the state regardless of whether the button was clicked, because we store that information in the mouse state. Butl well talk about that in a few mins